src/syscall_handler.c

```diff
@@ -23,6 +24,31 @@ uint32_t sys_testcall(uint32_t code) {
     return 0;
 }
 
+uint32_t sys_open(uint32_t path_intptr, uint32_t mode_intptr) {
+    return fs_sys_open((const char *)path_intptr, (const char *)mode_intptr);
```

Comment:
How many levels of function wrapping do we want? Can't we just call the fs_sys* in the switch for the handler, rather than putting another function on the stack just for casting?